services/app-api/src/postgres/contractAndRates/findAllRatesWithHistoryBySubmitInfo.ts

```diff
@@ -0,0 +1,49 @@
+import type { PrismaTransactionType } from '../prismaTypes'
+import type { RateType } from '../../domain-models/contractAndRates'
+import { parseRateWithHistory } from './parseRateWithHistory'
+import { includeFullRate } from './prismaSubmittedRateHelpers'
+
+type RateOrErrorType = {
+    rateID: string
+    rate: RateType | Error
+}
+
+type RateOrErrorArrayType = RateOrErrorType[]
+
+async function findAllRatesWithHistoryBySubmitInfo(
+    client: PrismaTransactionType
+): Promise<RateOrErrorArrayType | Error> {
+    try {
+        const rates = await client.rateTable.findMany({
+            where: {
+                revisions: {
+                    some: {
+                        submitInfo: {
+                            isNot: null,
+                        },
+                    },
+                },
+                stateCode: {
+                    not: 'AS', // exclude test state as per ADR 019
+                },
+            },
+            include: {
+                ...includeFullRate,
+                state: true,
+            },
+        })
+
+        const parsedRatesOrErrors: RateOrErrorArrayType = rates.map((rate) => ({
```

Comment:
The type here is not `Rate | Errors` it is

```
{
    rateID: string
    rate: RateType | Error
}[]
```
and the reason its returned (I grabbed this from findAllContracts code) makes sense to me - looks like we want to return the keys for all possible records so that we can clearly send back of list of which records errors and which didn't for display in the table. 

@macrael is there a change you want here with the types or the approach - I'm happy to make it in both places.